app/Filament/Resources/UserUsernameResource.php

```diff
@@ -0,0 +1,176 @@
+                    ->url(fn (UserUsername $record) => UserResource::getUrl('view', ['record' => $record->user->display_name]))
+                    ->extraAttributes(['class' => 'underline'])
+                    ->openUrlInNewTab()
+                    ->searchable()
+                    ->sortable(),
+
+                Tables\Columns\TextColumn::make('user.display_name')
+                    ->label('Current Username')
+                    ->url(fn (UserUsername $record) => UserResource::getUrl('view', ['record' => $record->user->display_name]))
+                    ->extraAttributes(['class' => 'underline'])
+                    ->openUrlInNewTab()
+                    ->searchable()
+                    ->sortable(),
+
+                Tables\Columns\TextColumn::make('username')
+                    ->label('Requested New Username')
+                    ->searchable()
+                    ->sortable(),
+
+                Tables\Columns\TextColumn::make('created_at')
+                    ->label('Requested At')
+                    ->dateTime()
+                    ->sortable(),
+
+                Tables\Columns\TextColumn::make('status')
+                    ->label('Status')
+                    ->state(fn (UserUsername $record): string => match (true) {
+                        $record->is_approved => 'Approved',
+                        $record->is_denied => 'Denied',
+                        default => 'Pending',
+                    })
+                    ->icon(fn (UserUsername $record): string => match (true) {
+                        $record->is_approved => 'heroicon-o-check-circle',
+                        $record->is_denied => 'heroicon-o-x-circle',
+                        default => 'heroicon-o-clock',
+                    })
+                    ->color(fn (UserUsername $record): string => match (true) {
+                        $record->is_approved => 'success',
+                        $record->is_denied => 'danger',
+                        default => 'warning',
+                    }),
+            ])
+            ->defaultSort('created_at', 'desc')
+            ->filters([
+                Tables\Filters\SelectFilter::make('status')
+                    ->options([
+                        'pending' => 'Pending',
+                        'approved' => 'Approved',
+                        'denied' => 'Denied',
+                    ])
+                    ->query(function ($query, $state) {
+                        if (!isset($state['value'])) {
+                            return $query;
+                        }
+
+                        return match ($state['value']) {
+                            'pending' => $query->pending(),
+                            'approved' => $query->approved(),
+                            'denied' => $query->denied(),
+                            default => $query,
+                        };
+                    })
+                    ->default('pending'),
```

Comment:
It seems unlikely, but if a request is not addressed within 30 days, it falls off the default sort and the user is allowed to make new requests. The existing request is still present and can be accessed (and approved/denied) by removing this filter.

To test: update the `created_at` of a pending request to be more than 30 days ago.

Reply:
In latest, the table will only show non-expired rows.